Makefile

```diff
@@ -152,7 +152,7 @@ test_compilation:
 	$(MAKE) -C tests/spark test
 	$(MAKE) -C examples/apps/ping build
 	$(MAKE) -C examples/apps/dhcp_client build
-	$(PYTEST) -m "compilation and not verification" tests
```

Comment:
This change excludes `tests/tools/check_doc_tests` which should be included.

Reply:
I think all tests in `tests/tools` and `tests/ide` are not executed anymore (in the CI). Maybe we should add separate make targets for these tests?

Reply:
I restructured the `test_python*` targets a bit. I hope I didn't miss anything,